containers/daap-delete-data-product/tests/unit/mock_test.py

```diff
@@ -0,0 +1,161 @@
+    "dpiaRequired": False,
+    "schemas": ["schema0", "schema1", "schema2"],
+}
+
+test_schema: dict[str, Any] = {
+    "tableDescription": "table has schema to pass test",
+    "columns": [
+        {
+            "name": "col_1",
+            "type": "bigint",
+            "description": "ABCDEFGHIJKLMNOPQRSTUVWXY",
+        },
+        {"name": "col_2", "type": "tinyint", "description": "ABCDEFGHIJKL"},
+        {
+            "name": "col_3",
+            "type": "int",
+            "description": "ABCDEFGHIJKLMNOPQRSTUVWX",
+        },
+        {"name": "col_4", "type": "smallint", "description": "ABCDEFGHIJKLMN"},
+    ],
+}
+
+
+class TestRemoveAllVersions:
+    @pytest.fixture(autouse=True)
+    def setup(
+        self,
+        s3_client,
+        glue_client,
+        data_product_name,
+        data_product_versions,
+        create_metadata_bucket,
+    ):
+        for version in data_product_versions:
+            s3_client.put_object(
+                Body=json.dumps(test_metadata_with_schemas),
+                Bucket=os.getenv("METADATA_BUCKET"),
+                Key=f"{data_product_name}/{version}/metadata.json",
+            )
+        for i in range(3):
+            for version in data_product_versions:
+                s3_client.put_object(
+                    Body=json.dumps(test_schema),
+                    Bucket=os.getenv("METADATA_BUCKET"),
+                    Key=f"{data_product_name}/{version}/schema{i}/schema.json",
+                )
+
+    def test_success(
+        self,
+        s3_client,
+        create_glue_tables,
+        create_raw_and_curated_data,
+        data_product_name,
+        glue_client,
+        event,
+        fake_context,
+    ):
+        # Assert we have the correct number of database tables created
+        tables = glue_client.get_tables(DatabaseName=data_product_name)
+        assert len(tables["TableList"]) == 3
+
+        # Assert we have the correct number of metadata files to begin with
+        prefix = f"{data_product_name}/"
+        response = s3_client.list_objects_v2(
+            Bucket=os.getenv("METADATA_BUCKET"),
+            Prefix=prefix,
+        )
+        assert response.get("KeyCount") == 12
+
+        # Assert we have the correct number of metadata files to begin with
+        prefix = f"raw/{data_product_name}/"
+        response = s3_client.list_objects_v2(
+            Bucket=os.getenv("RAW_DATA_BUCKET"),
+            Prefix=prefix,
+        )
+        assert response.get("KeyCount") == 30
+
+        # Assert we have the correct number of metadata files to begin with
+        prefix = f"curated/{data_product_name}/"
+        response = s3_client.list_objects_v2(
+            Bucket=os.getenv("CURATED_DATA_BUCKET"),
+            Prefix=prefix,
+        )
+        assert response.get("KeyCount") == 30
+
+        ######################################################################
+        with patch("glue_utils.glue_client", glue_client):
```

Comment:
Just an idea - another way of testing this would be to mock out the VersionCreator object. Then this test could verify that the `remove_all_versions_of_a_data_product` method is called, and the unit test for that method can verify that this method actually does the right thing in glue.

I think it's fine either way, it just depends how much of the integration you want to cover in this specific test

Reply:
Thanks, ill incorporate this going forward in my testing.